DOC-3215: Docs changes for new onboarding feature. (#3726)
tinymce/tinymce-docs: modules/ROOT/pages/cloud-troubleshooting.adoc

NOTE: The wording of the notifications shown here may differ from the actual notifications from {cloudname}.

'''
[[invalid-api-key-cloud-troubleshooting]]
== Invalid API Key

[[A-valid-API-key-is-required-to-continue-using-TinyMCE.-Please-alert-the-admin-to-check-the-current-API-key]]
=== "A valid API key is required to continue using {productname}. **Please alert the admin** to check the current API key. xref:invalid-api-key.adoc[Click here to learn more.]"

==== Cause

This notification is shown when either:

* An `+apiKey+` is not provided in the script tag,
* An empty or missing `+apiKey+` is provided, or
* The `+apiKey+` provided cannot be found on the {cloudname} server.

The `+apiKey+` must be:

* An exact length,
* comprised of certain characters, and
* created with a {cloudname} account on the link:{accountsignup}/[{accountpage} page].

For example:

[source,html,subs="attributes+"]
----
<script src="https://cdn.tiny.cloud/1/abcd1234/tinymce/{productmajorversion}/tinymce.min.js" referrerpolicy="origin"></script>
----

> In this case, the editor will be in a **read-only** state, and the user will see a **"Finish setting up"** dialog prompting them to add a valid API key, configure their domain, and explore premium features.
> 

Reviewer: ```suggestion
In this case, the editor will be in a **read-only** state, and a notification appears stating: A valid API key is required to continue using TinyMCE. Please alert the admin to check the current API key. Click here to learn more.
```

[[A-valid-API-key-is-required-to-continue-using-TinyMCE.-Please-alert-the-admin-to-check-the-current-API-key]]
=== "A valid API key is required to continue using {productname}. **Please alert the admin** to check the current API key. xref:invalid-api-key.adoc[Click here to learn more.]"

==== Cause "no-api-key"

This notification is *only shown* when either:

image:onboarding/onboarding-no-api-key-notification.png[no-api-key-notificaiton-banner, width=800, height=auto]

> * An `+apiKey+` is not provided in the script tag, which triggers the **"Finish setting up"** dialog.
> * When `+no-api-key+` is provided as the API key inside the script tag.

Author: ```suggestion
* An `+apiKey+` is not provided in the script tag, or when `+no-api-key+` is provided as the API key.
```

'''

> [[trial-banner-exploration]]
> == Trial Banner - Explore Features (Valid Trial API Key)
> 
> === "Try out paid features! Explore all of the trial benefits including Premium plugins."
> 
> ==== Cause
> 
> This banner appears when the API key being used is associated with a trial account. The editor is fully functional but includes a banner promoting premium features such as `PowerPaste`, `Export to Word`, and `Templates`.
> 
> 
> image::onboarding/onboarding-trial-explore-features.png[trial-explore-features-notification-banner, width=400, height=auto]
> 
> ==== Solution
> 
> To disable this banner:
> 
> Set the `+onboarding+` option to `+false+` in your {productname} configuration:
> 
> [source,javascript]
> ----
> tinymce.init({
> selector: 'textarea',
> onboarding: false, // Disables trial banner
> plugins: 'lists link image table',
> toolbar: 'undo redo | bold italic | alignleft aligncenter alignright | bullist numlist | link image',
> });
> ----
> 
> Alternatively, upgrading your API key to a paid plan will remove the trial messages.
> 

Author: Suggested we remove this
```suggestion
```
